### PR TITLE
Improve Black Friday

### PR DESCRIPTION
**Pull request by EpokK**
Capture d’écran 2019-11-28 à 13 47 52

- New component added: `Countdown`

Fix https://gitlab.protontech.ch/ProtonVPN/web/protonvpn-settings/issues/51
Manager Delong
Test server https://vpnsettings.protonmail.blue (`blue` API)

### CODE REVIEW

File: components/countdown/Countdown.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    const minutes = Math.floor((diff % HOUR) / MINUTE);
+    const seconds = Math.floor((diff % MINUTE) / SECOND);
+
+    return (
+        <>
+            {[
+                c('Countdown unit').t`${days} days`,
+                c('Countdown unit').t`${hours} hours`,
+                c('Countdown unit').t`${minutes} minutes`,
+                c('Countdown unit').t`${seconds} seconds`
```
Comment (econdepe) [inline components/countdown/Countdown.js]: Maybe we should take into account plurals here?

File: containers/payments/subscription/BlackFridayModal.js [hunk 2]
```diff
@@ -32,19 +40,19 @@ const BlackFridayModal = ({ bundles = [], onSelect, ...rest }) => {
 
     const BILLED_DESCRIPTION = ({ cycle, amount, notice }) =>
         ({
-            [MONTHLY]: c('Title').jt`Billed as ${amount} (${notice})`,
-            [YEARLY]: c('Title').jt`Billed as ${amount} (${notice})`,
-            [TWO_YEARS]: c('Title').jt`Billed as ${amount} (${notice})`
+            [MONTHLY]: c('Title').jt`Billed as ${amount} ${NOTICES[notice]} for 1 month`,
```
Comment (econdepe) [inline containers/payments/subscription/BlackFridayModal.js]: I would have the asterisks by the end: `Billed as ${amount} for 1 month ${NOTICES[notice]}`. I think they are also at the end in the proposal

**Comment by econdepe** [comment]
Mmhh, what I said above is probably not a good idea given the async nature of the setState operations. Probably what you wrote is more stable

File: components/countdown/Countdown.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    const hours = Math.floor((diff % DAY) / HOUR);
+    const minutes = Math.floor((diff % HOUR) / MINUTE);
+    const seconds = Math.floor((diff % MINUTE) / SECOND);
+
+    return (
+        <>
+            {[
+                c('Countdown unit').ngettext(msgid`${days} day`, `${days} days`, days),
+                c('Countdown unit').ngettext(msgid`${hours} hour`, `${hours} hours`, days),
+                c('Countdown unit').ngettext(msgid`${minutes} minute`, `${minutes} minutes`, days),
```
Comment (econdepe) [inline components/countdown/Countdown.js]: When `day` reaches 0, do we want to still display it? Same question for hours, when `hour` and `day` reach 0, and for minutes, when `day`, 'hour` and `minutes` reach 0

Comment (EpokK) [reply components/countdown/Countdown.js]: Yes we want

**Pull request merged**